ppb/systems.py

```diff
@@ -55,9 +63,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     def on_update(self, update, signal):
         for pygame_event in pygame.event.get():
-            ppb_event = self.event_map.get(pygame_event.type)
-            if ppb_event is not None:
-                signal(getattr(self, ppb_event)(pygame_event, update.scene))
+            methname = self.event_map.get(pygame_event.type)
+            if methname is not None:
+                ppbevent = getattr(self, methname)(pygame_event, update.scene)
+                if ppbevent:
```

Comment:
This should never happen. If it's in the method map, it's on the class, and if it's not, that's on us.

Reply:
That's not what this tests. This tests if the pygame event actually produced a ppb event.